src/evaluation/evaluator.h

```diff
@@ -33,22 +59,26 @@ class Evaluator {
 
         const uint64_t hash = engine::generateHashKey(board);
 
-        m_searcher.setHashKey(hash);
+        for (auto& searcher : m_searchers) {
+            searcher->setHashKey(hash);
+        }
 
         return scanForBestMove(depthInput.value_or(s_maxSearchDepth), board);
     }
 
-    constexpr bool startPondering(const BitBoard& board)
+    constexpr bool startPondering(const BitBoard& board, uint8_t numSearchers)
     {
         if (m_isPondering || !m_ponderingEnabled)
             return false;
 
         m_isPondering = true;
-        return getBestMoveAsync(board);
+        return getBestMoveAsync(board, numSearchers);
     }
 
-    constexpr bool getBestMoveAsync(const BitBoard& board, std::optional<uint8_t> depthInput = std::nullopt)
+    constexpr bool getBestMoveAsync(const BitBoard& board, uint8_t numSearchers, std::optional<uint8_t> depthInput = std::nullopt)
```

Comment:
I don't think it's necessary to pass the searchers as an argument.
We had a talk about it and the prior implementation (your first draft) it was the consumers responsibility to resize, check etc.

I think the searchers should only be resized from the uci option which will happen at the start and nowhere else :)